.gitignore

```diff
@@ -14,3 +14,4 @@ generated/
 /debug/
 /antlr-3.4
 deps/
+/.vscode/
```

Comment:
This would go on a separate PR, it is not related to optimization.
Meanwhile, you can exclude these locally:
https://stackoverflow.com/questions/5724455/can-i-make-a-user-specific-gitignore-file